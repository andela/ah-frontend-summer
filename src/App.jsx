import React, { Component } from 'react';
import './styles/App.scss';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h5>Welcome to Authors Haven</h5>
      </div>
    );
  }
}

export default App;
