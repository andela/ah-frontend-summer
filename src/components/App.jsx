import React from 'react';

const App = () => {
    return (
        <div className="App">
            <h5>Welcome to Authors Haven</h5>
        </div>
    );
};

export default App;
