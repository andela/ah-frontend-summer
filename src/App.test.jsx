import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from 'redux';
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import reducer from "./store/reducers";
import Routes from './routes';
import App from './components/App';


const mockStore = createStore(
    reducer, // , pass in the main reducer
    applyMiddleware(thunk) // , apply any middleware we are making use of
);

describe('App rendering tests', () => {
    it('app renders without crashing', () => {
        shallow(<App />);
    });

    it('renders consistently', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders without crashing', () => {
        const wrapper = shallow(<Routes />);
        expect(wrapper).toMatchSnapshot();
    });

    it('routes render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Provider store={mockStore}><Routes /></Provider>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
});
