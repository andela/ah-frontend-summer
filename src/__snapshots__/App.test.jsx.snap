// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App rendering tests renders consistently 1`] = `
<div
  className="app"
>
  <h5>
    Welcome to Authors Haven
  </h5>
</div>
`;

exports[`App rendering tests renders without crashing 1`] = `
<BrowserRouter>
  <div>
    <Switch>
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Connect(Layout)>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles/:slug"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles/create"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles/update/:slug"
        />
        <Route
          component={[Function]}
          path="/tags"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/email-verification/:token"
        />
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />
      </Connect(Layout)>
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`App rendering tests renders without crashing 2`] = `
<BrowserRouter>
  <div>
    <Switch>
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Connect(Layout)>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles/:slug"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles/create"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/articles/update/:slug"
        />
        <Route
          component={[Function]}
          path="/tags"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/email-verification/:token"
        />
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />
      </Connect(Layout)>
    </Switch>
  </div>
</BrowserRouter>
`;
